app/allocation/controllers/view.js

```diff
@@ -22,7 +27,7 @@ module.exports = function view(req, res) {
       emptySlots: movesWithoutPerson.length,
       filledSlots: moves
         .filter(move => move.person)
-        .map(presenters.moveToCardComponent()),
+        .map(move => ({ ...move, ...moveToCardComponent(move) })),
```

Comment:
This feels a little messy combining the the two properties back together. And could lead to unintended circumstances in the future of properties being overridden.

If, for this solution we need access to the original move I think it would be better to assign the result of the presenter to a separate property. 

Or even better split the original move and the presenter to separate properties for this collection that you send to the view.

Or even, even better. Send an object with the card result and only the properties you need:

```javascript
.map(move => {
  return {
    id: move.id,
    fullname: move.person.fullname,
    card: moveToCardComponent(move),
  }
})
```

Reply:
👍